Packs/BreachRx/Integrations/BreachRx/BreachRx.py

```diff
@@ -0,0 +1,301 @@
+        self.org_name = org_name
+
+        auth = HTTPBasicAuth(api_key, secret_key)
+
+        transport = RequestsHTTPTransport(
+            url=base_url,
+            auth=auth,
+            headers={"orgname": org_name},
+            timeout=60
+        )
+
+        self.client = Client(
+            transport=transport, fetch_schema_from_transport=True
+        )
+
+    def get_incident_severities(self):
+        return self.client.execute(get_incident_severities)["incidentSeverities"]
+
+    def get_incident_types(self):
+        return self.client.execute(get_incident_types)["types"]
+
+    def create_incident(self, name: Optional[str], description: Optional[str]):
+        severities = self.get_incident_severities()
+        types = self.get_incident_types()
+
+        params = {
+            "severity": severities[0]["name"],
+            "name": name,
+            "type": types[0]["name"],
+            "description": description
+        }
+        return self.client.execute(create_incident_mutation, params)["createIncident"]
+
+    def get_incident(self, name: Optional[str], identifier: Optional[str]):
+        params = {
+            "name": name,
+            "identifier": identifier
+        }
+        results = self.client.execute(get_incident_by_name, params)['incidents']
+
+        if results:
+            return results.pop()
+        else:
+            return None
+
+    def get_actions_for_incident(self, incident_id):
+        params = {
+            "incidentId": incident_id
+        }
+
+        return self.client.execute(get_actions_for_incident, params)["actions"]
+
+
+def test_module(client: BreachRxClient):
+    try:
+        client.get_incident_severities()
+        return "ok"
+    except Exception:
+        return "Authorization Error: make sure your API Key and Secret Key are correctly set"
+
+
+def create_incident_command(
+    client: BreachRxClient,
+    incident_name: str = demisto.incident().get("name"),
+    description: str = None
+) -> CommandResults:
+    if not description:
+        description = (
+            f"""An Incident copied from the Palo Alto Networks XSOAR platform.
+            <br>
+            <br>
+            XSOAR Incident Name: {demisto.incident().get('name')}"""
+        )
+
+    response = client.create_incident(incident_name, description)
+
+    incident_name = response["name"]
+
+    return CommandResults(
+        outputs_prefix="BreachRx.Incident",
+        outputs_key_field="id",
+        outputs=response,
+        raw_response=response,
+        readable_output=f"Incident created with name={incident_name}."
+    )
+
+
+def get_incident_actions_command(
+    client: BreachRxClient,
+    incident_name: str = None,
+    incident_identifier: str = None
+) -> Union[CommandResults, str]:
+    incident_id = demisto.dt(demisto.context(), 'BreachRx.Incident.id')
+
+    if not incident_name:
+        incident_name = demisto.dt(demisto.context(), 'BreachRx.Incident.name')
```

Comment:
Why do we want to extract the name and id from the context if not supplied in the command args?

Reply:
The team working on the privacy Incident on the BreachRx platform often isn't the same team as the one using Cortex XSOAR, so in order to allow the team using XSOAR to "check in" on the privacy team without having to memorize the Incident information, we store that info on the Incident's context.

In other words, we do not want the team using XSOAR to have to supply the BreachRx incident information every time they want to view the Actions being worked on by the privacy team.